### ISSUE TITLE
Bug: Empty Entity date attributes receive current timestamp when accessed

### ISSUE THREAD

**Issue by JanZelenka** [codeigniter4/CodeIgniter4#3251]
I have the following scenario:

* Using the Entity class
* Setting the $dates property to an array of attributes that are supposed to be of type DateTime
* On loading the Entity data from the database, the dates have no value
* When trying to get the value, *__get()* doesn't check for an empty value, calls the mutateDate method, which in turn end up calling *Time::parse()* resulting in the current timestamp.

*CodeIgniter 4 version:* 4.0.3
*Affected module:* Entity class
*PHP version:* 7.2

**Comment by Instrye** [comment]
This is expected, because `new DateTime('')` return current time. I think this is a problem with your data rather than a problem with the framework.

**Comment by JanZelenka** [comment]
Thanks for the feedback but I still believe this is a framework issue. Imagine this scenario:

I have a table _Process_ with column _startedOn_ in my database. When this column is empty, it indicates a process that hasn't started yet.
In CodeIgniter I create an Entity class _Process_. I load data of a specific process into _$CurrentProcess_ of type _Process_ and try this:
`if ( empty( $CurrentProcess->starterdOn ) ) { $CurrentProcess->start() }`

My processes never get started because even though in the database they have no startup date specified, CodeIgniter substitutes my empty value with the current timestamp.

Bottom line, if a column is empty in the database, the respective attribute in your Entity must be empty as well.

**Comment by Instrye** [comment]
I will submit a PR based on your suggestion, but whether to merge or not depends on the judgment of the administrator.

**Comment by MGatner** [comment]
I think the expectation is that you would use `null` in your database, so the result wouldn't be a string. While I don't know what one would expect from mutating `""` I agree getting back the current time stamp is unexpected.

**Comment by JanZelenka** [comment]
Yeah, that would indeed be the expectation - except my database engine (FileMaker) doesn't recognize `null`, `""` is its equivalent of `null`

**Comment by michalsn** [comment]
Well... I believe that in the Entity class we have a special use for the empty string and it should not be changed.

@JanZelenka If your database is specific and there is no support for `null` values then you should probably extend the Entity class and make changes just for your own use.

An empty string should still be supported and return the current date - this is expected behavior.
